### PR TITLE
Create instance db

### PR DESCRIPTION
**Pull request by antho1404**
Add instance struct and instance DB

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -20,6 +20,7 @@ const (
 
 	serviceDBVersion   = "v2"
 	executionDBVersion = "v1"
+	instanceDBVersion  = "v1'"
```
Comment (NicolasMahe) [inline config/config.go]:
```suggestion
	instanceDBVersion  = "v1"
```

Comment (NicolasMahe) [reply config/config.go]: why is it resolved but not fixed?

File: database/instance_db.go [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+	// encode service
+	b, err := d.marshal(i)
+	if err != nil {
+		tx.Discard()
+		return err
+	}
+
+	// save instance with hash.
+	if err := tx.Put([]byte(i.Hash), b, nil); err != nil {
```
Comment (antho1404) [inline database/instance_db.go]: @krhubert https://github.com/mesg-foundation/engine/pull/1037#discussion_r291853397
> the key can be saved with a single call without using transaction

File: database/instance_db.go [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+	return &s, nil
+}
+
+// Get retrives instance by instance hash.
+func (d *LevelDBInstanceDB) Get(hash string) (*instance.Instance, error) {
+	tx, err := d.db.OpenTransaction()
+	if err != nil {
+		return nil, err
+	}
+	b, err := tx.Get([]byte(hash), nil)
```
Comment (antho1404) [inline database/instance_db.go]: @krhubert https://github.com/mesg-foundation/engine/pull/1037#discussion_r291853412
> the key can be get with a single call without using transaction

File: database/instance_db.go [hunk 4]
```diff
@@ -0,0 +1,104 @@
+package database
```
Comment (antho1404) [inline database/instance_db.go]: @krhubert https://github.com/mesg-foundation/engine/pull/1037#discussion_r291853650
> There are no tests for this package

File: instance/instance.go [hunk 5]
```diff
@@ -0,0 +1,9 @@
+package instance
+
+// Instance is an instance of a service.
+// This contains a reference to a service that is running.
+// Multiple instances can run for the same service as long as they have different configurations
+type Instance struct {
+	Hash        string `hash:"-"`
```
Comment (antho1404) [inline instance/instance.go]: @krhubert https://github.com/mesg-foundation/engine/pull/1037#discussion_r291853448
> instance tags are not needed here (structhash is not used on the struct)

**Comment by ilgooz** [comment]
are you sure that writes are atomic without using transactions? it doesn't worth to rm transactions now since db is going to changed soon, there is no reason to introduce any instability in this period.

**Comment by ilgooz** [comment]
@antho1404 this is blocking, can you give the last decision about transactions please?

**Comment by antho1404** [comment]
I don't see the issue with the transaction, this is already used in the execution db so I don't see why there is an issue with that

**Pull request merged**